### PR TITLE
Use golang 1.17

### PR DESCRIPTION
**Pull request by pradithya**
**What this PR does / why we need it**:

Some package in https://github.com/gojek/merlin/pull/221 requires newer version of golang (i.e. 1.15) . This PR is to update golang version in CI, docker, and development environment to 1.17

**Which issue(s) this PR fixes**:

None

**Does this PR introduce a user-facing change?**:
None

**Checklist**

- [x] Added unit test, integration, and/or e2e tests
- [x] Tested locally
- [x] Updated documentation

### CODE REVIEW

File: api/go.mod [hunk 1]
```diff
@@ -81,6 +73,121 @@ require (
 	sigs.k8s.io/yaml v1.2.0
 )
 
+require (
```
Comment (ariefrahmansyah) [inline api/go.mod:76]: Why is there two require blocks?

Comment (ariefrahmansyah) [reply api/go.mod:76]: You might need to run `go mod tidy`

Comment (pradithya) [reply api/go.mod:76]: It's because in 1.17 they implements https://go-review.googlesource.com/c/go/+/325922/

Comment (ariefrahmansyah) [reply api/go.mod:76]: I see. #TIL

**Comment by ariefrahmansyah** [comment]
LGTM.

However, you need to update Makefile for remaining test recipes: https://github.com/gojek/merlin/blob/main/Makefile#L71-L81
You need to add `-gcflags=all=-d=checkptr=0` flag.

**Review by ariefrahmansyah — APPROVED** [review]

**Pull request merged**